pySDC/implementations/problem_classes/Burgers.py

```diff
@@ -0,0 +1,315 @@
+
+        super().__init__(bases=bases, components=components, spectral_space=False, **kwargs)
+
+        self.x = self.get_grid()[0]
+
+        # prepare matrices
+        Dx = self.get_differentiation_matrix(axes=(0,))
+        I = self.get_Id()
+
+        T2U = self.get_basis_change_matrix(conv=mode)
+
+        self.Dx = Dx
+
+        # construct linear operator
+        L_lhs = {'u': {'ux': -epsilon * (T2U @ Dx)}, 'ux': {'u': -T2U @ Dx, 'ux': T2U @ I}}
+        self.setup_L(L_lhs)
+
+        # construct mass matrix
+        M_lhs = {'u': {'u': T2U @ I}}
+        self.setup_M(M_lhs)
+
+        # boundary conditions
+        self.add_BC(component='u', equation='u', axis=0, x=1, v=BCr, kind='Dirichlet')
+        self.add_BC(component='u', equation='ux', axis=0, x=-1, v=BCl, kind='Dirichlet')
+        self.setup_BCs()
+
+    def u_exact(self, t=0, *args, **kwargs):
+        me = self.u_init
+
+        # x = (self.x + 1) / 2
+        # g = 4 * (1 + np.exp(-(4 * x + t)/self.epsilon/32))
+        # g_x = 4 * np.exp(-(4 * x + t)/self.epsilon/32) * (-4/self.epsilon/32)
+
+        # me[0] = 3./4. - 1./g
+        # me[1] = 1/g**2 * g_x
+
+        # return me
+
+        if t == 0:
+            me[self.index('u')][:] = ((self.BCr + self.BCl) / 2 + (self.BCr - self.BCl) / 2 * self.x) * np.cos(
+                self.x * np.pi * self.f
+            )
+            me[self.index('ux')][:] = (self.BCr - self.BCl) / 2 * np.cos(self.x * np.pi * self.f) + (
+                (self.BCr + self.BCl) / 2 + (self.BCr - self.BCl) / 2 * self.x
+            ) * self.f * np.pi * -np.sin(self.x * np.pi * self.f)
+        elif t == np.inf and self.f == 0 and self.BCl == -self.BCr:
+            me[0] = (self.BCl * np.exp((self.BCr - self.BCl) / (2 * self.epsilon) * self.x) + self.BCr) / (
+                np.exp((self.BCr - self.BCl) / (2 * self.epsilon) * self.x) + 1
+            )
+        else:
+            raise NotImplementedError
+
+        return me
+
+    def eval_f(self, u, *args, **kwargs):
+        f = self.f_init
+        iu, iux = self.index('u'), self.index('ux')
+
+        u_hat = self.transform(u)
+
+        Dx_u_hat = self.u_init_forward
+        Dx_u_hat[iux] = (self.Dx @ u_hat[iux].flatten()).reshape(u_hat[iu].shape)
+
+        f.impl[iu] = self.epsilon * self.itransform(Dx_u_hat)[iux].real
+        f.expl[iu] = -u[iu] * u[iux]
+        return f
+
+    def get_fig(self):  # pragma: no cover
+        """
+        Get a figure suitable to plot the solution of this problem
+
+        Returns
+        -------
+        self.fig : matplotlib.pyplot.figure.Figure
+        """
+        import matplotlib.pyplot as plt
+
+        plt.rcParams['figure.constrained_layout.use'] = True
+        self.fig, axs = plt.subplots()
+        return self.fig
+
+    def plot(self, u, t=None, fig=None, comp='u'):  # pragma: no cover
+        r"""
+        Plot the solution. Please supply a figure with the same structure as returned by ``self.get_fig``.
```

Comment:
the `Please supply a figure` .... should be in put in the documentation of `fig` below, explicitly mentioning that if not provided, the one provided by `self.get_fig` is used.